packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2388,7 +2387,7 @@ export class TransactionController extends BaseControllerV1<
    * @param transactionMeta - Nominated external transaction to be added to state.
    */
   private addExternalTransaction(transactionMeta: TransactionMeta) {
-    const chainId = this.getChainId();
+    const { chainId } = transactionMeta;
```

Comment:
I'm realizing that introducing these separate from where we are introducing the true multichain logic is that [later steps in the flow](https://github.com/MetaMask/core/blob/9ff9e45853d6c775e36b7360deb96052eba1c214/packages/transaction-controller/src/TransactionController.ts#L1229) here (confirmExternalTransaction, as just one example) rely on the globally selected chainId matching the chainId of the transaction being processed... so even though in some sense it doesn't really makes sense that confirmExternalTransaction should ever be allowed to be called with a transaction that doesn't match the currently globally selected chainId we are now introducing the possibility that a transaction gets added with a chainId passed in the transactionMeta but then [gets a postTransactionBalances added to it](https://github.com/MetaMask/core/blob/9ff9e45853d6c775e36b7360deb96052eba1c214/packages/transaction-controller/src/TransactionController.ts#L2663) for a different chain since ethQuery will be pointed to the globally selected chain